File: android/app/src/main/kotlin/com/hyequest/audiorecorder/data/audiorecorder/RecordWaveformModel.kt
package com.hyequest.audiorecorder.data.audiorecorder

data class RecordWaveformModel(val timestamp: Double, val data: List<Double>) {

    fun toMap(): Map<String, Any> {
        return mapOf(
            "timestamp" to timestamp,
            "data" to data
        )
    }
}

File: ios/Runner/data/AudioRecorder/RecordWaveformModel.swift
//
//  RecordWaveformModel.swift
//  Runner
//
//  Created by Hyebreed on 10/08/2025.
//

class RecordWaveformModel {
    let timestamp: TimeInterval
    let data: [Double]
    
    init(timestamp: TimeInterval, data: [Double]) {
        self.timestamp = timestamp
        self.data = data
    }
    
    func toDictionary() -> [String: Any]{
        return [
            "timestamp": timestamp,
            "data": data
        ]
    }
}
